cargo_vendor

```diff
@@ -88,12 +89,18 @@ def run_cargo(runDirectory, command, argsList=[]):
         return None
 
 
-def cargo_vendor(appDirectory, update, outdir, tag, argsList=[]):
+def cargo_vendor(appDirectory, update, outdir, tag, cargotoml=None, argsList=[]):
     if update:
         log.info("Updating deps before vendor")
         run_cargo(appDirectory, "update")
     log.info(f"Vendoring Cargo.toml deps to {appDirectory}/vendor")
-    output = run_cargo(appDirectory, "vendor", argsList + ["--locked", "--", "vendor"])
+    sync_args = []
+    if cargotoml:
+        for toml in cargotoml:
+            log.info(f"Adding path '{toml}' to sync options")
```

Comment:
Not sure this is right, I think the first needs to be --manifest-path, and then all the subsequent are --sync? 

I'm guessing you tested this and it works?

Reply:
I think from what I understood from the https://doc.rust-lang.org/cargo/commands/cargo-vendor.html?highlight=sync#vendor-options, it will add the manifest path automatically as
described in the Manifest Option:

> By default, Cargo searches for the Cargo.toml file in the current directory or any parent directory.

So default behavior will just add manifest path on the root of the project while subsequent syncs are added. I think the good thing about sync is that you can just even specify the root Cargo.toml (which I don't think anyone should do since there is already default behavior).

Reply:
maybe a `manifestpath` option should be considered if we are still confused by this 😅

Reply:
Okay, but if someone has a project with a single cargo.toml, and adds it then cargo will try to search for another, non-existant cargo.toml.

I think we need to make it so that the first one is manifest path so people with this option as a single value can just nominate their cargo.toml rather than rely on autosearch, and then all subsequent listed cargo.toml are --sync.

Reply:
Alrighty. Should I also update the README so people will expect that the first sync is for the manifest path? or should a separate option `manifestpath` be added since if it does not exist, it will default to cargo's manifestpath behavior?

Reply:
I think it should just be the first one in the list is implied as the manifest path.

Reply:
It seems the documentation of `manifest-path` is inconsistent. It should be `path/to/toml` instead of `path/relative/totoml`. Tested it with cargo. I will push a commit soon.

Reply:
We need to use the relative path though in this library though.